client/asset/eth/nodeclient_harness_test.go

```diff
@@ -143,65 +137,69 @@ func newRedeem(secret, secretHash [32]byte) *asset.Redemption {
 	}
 }
 
-func waitForMinedSimnet(t *testing.T, nc *nodeClient, timeLimit time.Duration, waitTimeLimit bool) error {
+// waitForReceipt waits for a tx. This is useful on testnet when a tx may be "missing"
+// due to reorg. Wait for a few blocks to find the main chain and hopefully our tx.
+func waitForReceipt(t *testing.T, nc *nodeClient, tx *types.Transaction) (*types.Receipt, error) {
 	t.Helper()
-	err := exec.Command("geth", "--datadir="+alphaNodeDir, "attach", "--exec", "miner.start()").Run()
-	if err != nil {
-		return err
-	}
-	defer func() {
-		_ = exec.Command("geth", "--datadir="+alphaNodeDir, "attach", "--exec", "miner.stop()").Run()
-	}()
-	timesUp := time.After(timeLimit)
-out:
+	hash := tx.Hash()
+	// Waiting as much as five blocks.
+	timesUp := time.After(5 * secPerBlock)
 	for {
 		select {
 		case <-ctx.Done():
-			return ctx.Err()
-		case <-timesUp:
-			return errors.New("timed out")
+			return nil, ctx.Err()
 		case <-time.After(time.Second):
-			txsa, err := nc.pendingTransactions()
-			if err != nil {
-				return fmt.Errorf("initiator pendingTransactions error: %v", err)
-			}
-			txsb, err := participantEthClient.pendingTransactions()
+			receipt, err := nc.transactionReceipt(ctx, hash)
 			if err != nil {
-				return fmt.Errorf("participant pendingTransactions error: %v", err)
-			}
-			if len(txsa)+len(txsb) == 0 {
-				break out
+				if errors.Is(err, asset.CoinNotFoundError) {
+					continue
+				}
+				return nil, err
 			}
+			spew.Dump(receipt)
+			return receipt, nil
+		case <-timesUp:
+			return nil, errors.New("wait for receipt timed out, txn might be missing due to reorg, check the following tx for a bad nonce or low gas cap")
+			spew.Dump(tx)
```

Comment:
Unreachable.  Leftover debugging or you wanted this before the `return`?

Reply:
Doh. Yes. Moved.